src/hpx_wrap.cpp

```diff
@@ -0,0 +1,127 @@
+//  Copyright (c) 20018 Nikunj Gupta
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// The following implementation is only possible for Linux systems.
+#if defined(__linux__)
+
+// include_libhpx_wrap is a weak symbol which helps to determine the course
+// of function calls at runtime. It has a deafult value of 0 which
+// corresponds to the program's entry point being main().
+// It is overriden in hpx/hpx_main.hpp. Thus, inclusion of the header file
+// will change the program's entry point to HPX's own custom entry point
+// initialize_main. Subsequent calls before entering main() are handled
+// by this code.
+extern int include_libhpx_wrap;
+int include_libhpx_wrap __attribute__((weak)) = 0;
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/hpx.hpp>
+
+#include<vector>
+#include <iostream>
+
+////////////////////////////////////////////////////////////////////////////////
+// Function declarations
+//
+
+namespace hpx{
+    // Main entry point of HPX runtime system
+    extern int hpx_entry(int argc, char* argv[]);
+}
+
+// HPX's implented program's entry point
+extern int initialize_main(int argc, char** argv, char** envp);
+
+// Real libc function __libc_start_main
```

Comment:
Please add a comment with a link to the source code of the original libc implementation.